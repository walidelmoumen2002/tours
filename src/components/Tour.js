import React, { useState } from "react";

const Tour = ({ id, name, info, image, price, removeTour }) => {
  const [readMore, setReadMore] = useState(false);
  return (
    <article className="single-tour">
      <img src={image} alt={name} />
      <footer>
        <div className="tour-info">
          <h2>{name}</h2>
          <p className="tour-price">{price}$</p>
        </div>
        <p>
          {readMore ? info : info?.substring(0, 200)}...
          <button onClick={() => setReadMore(!readMore)} className="btninfo">
            {readMore ? "show less" : "show more"}
          </button>
        </p>
        <button onClick={() => removeTour(id)} className="removebtn ">
          Not interested
        </button>
      </footer>
    </article>
  );
};
export default Tour;
