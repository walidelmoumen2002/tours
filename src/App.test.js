import renderer from "react-test-renderer";
import Tour from "./components/Tour";

it("load and displays Tours", () => {
  const tour = {
    id: "rec6d6T3q5EBIdCfD",
    name: "Best of Paris in 7 Days Tour",
    info: "Paris is synonymous with the finest things that culture can offer — in art, fashion, food, literature, and ideas. On this tour, your Paris-savvy Rick Steves guide will immerse you in the very best of the City of Light: the masterpiece-packed Louvre and Orsay museums, resilient Notre-Dame Cathedral, exquisite Sainte-Chapelle, and extravagant Palace of Versailles. You'll also enjoy guided neighborhood walks through the city's historic heart as well as quieter moments to slow down and savor the city's intimate cafés, colorful markets, and joie de vivre. Join us for the Best of Paris in 7 Days!",
    image:
      "https://dl.airtable.com/.attachments/a0cd0702c443f31526267f38ea5314a1/2447eb7a/paris.jpg",
    price: "1,995",
  };
  const component = renderer.create(<Tour tour={tour} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
