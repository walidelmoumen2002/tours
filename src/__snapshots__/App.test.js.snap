// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load and displays Tours 1`] = `
<article
  className="single-tour"
>
  <img />
  <footer>
    <div
      className="tour-info"
    >
      <h2 />
      <p
        className="tour-price"
      >
        $
      </p>
    </div>
    <p>
      ...
      <button
        className="btninfo"
        onClick={[Function]}
      >
        show more
      </button>
    </p>
    <button
      className="removebtn "
      onClick={[Function]}
    >
      Not interested
    </button>
  </footer>
</article>
`;
